src/user/user.controller.ts

```diff
@@ -71,4 +83,23 @@ export class UserController {
       throw error;
     }
   }
+
+  @ApiOperation({ description: 'Rota para retornar dados cadastrais' })
+  @UseGuards(JwtAuthGuard)
+  @ApiBearerAuth()
+  @Get('me')
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'Dados cadastrais encontrados.',
+  })
+  @ApiResponse({
+    status: HttpStatus.UNAUTHORIZED,
+    description: 'Não foi encontrado um token de autenticação válido.',
+  })
+  async returnUserInfo(@Req() req: Request): Promise<UserInfoDTO> {
+    let token = req.headers.authorization;
+    token = token.toString().replace('Bearer ', '');
+    const payload = this.jwtService.decode(`${token}`);
```

Comment:
Faz assim para ficar com a tipagem correta:
```suggestion
    const token = req.headers.authorization.toString().replace('Bearer ', '');
    const user = this.jwtService.decode(token) as JWTUser;
```